omod/src/test/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResourceTest.java

```diff
@@ -0,0 +1,74 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+/* * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.idgen.rest.resource;
+
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.openmrs.api.context.Context;
+
+import org.junit.Before;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.idgen.rest.resource.AutoGenerationOptionResource;
+
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResourceTest;
+
+public class AutoGenerationOptionResourceTest extends BaseDelegatingResourceTest<AutoGenerationOptionResource, AutoGenerationOption> {
+	
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public AutoGenerationOption newObject() {
+		return Context.getService(IdentifierSourceService.class).getAutoGenerationOption(2);
+	}
+	
+	@Override
+	public void validateDefaultRepresentation() throws Exception {
+		super.validateDefaultRepresentation();
+		assertPropEquals("uuid", getUuidProperty());
+		assertPropEquals("location", getObject().getLocation().getUuid());
+		assertPropEquals("identifierType", getObject().getIdentifierType().getUuid());
+		assertPropEquals("source", getObject().getSource().getUuid());
+		assertPropEquals("manualEntryEnabled", getObject().isManualEntryEnabled());
+		assertPropEquals("automaticGenerationEnabled", getObject().isAutomaticGenerationEnabled());
+	}
+	
+	@Override
+	public void validateFullRepresentation() throws Exception {
+		super.validateFullRepresentation();
+		assertPropEquals("uuid", getUuidProperty());
+		assertPropEquals("location", getObject().getLocation().getUuid());
+		assertPropEquals("identifierType", getObject().getIdentifierType().getUuid());
+		assertPropEquals("source", getObject().getSource().getUuid());
+		assertPropEquals("manualEntryEnabled", getObject().isManualEntryEnabled());
+		assertPropEquals("automaticGenerationEnabled", getObject().isAutomaticGenerationEnabled());
+	}
+	
+	@Override
+	public String getDisplayProperty() {
+		return getObject().getIdentifierType() + " - " + getObject().getSource() + " - " + getObject().getLocation();
+	}
+	
+	@Override
+	public String getUuidProperty() {
+		return getObject().getIdentifierType().getUuid();
```

Comment:
As i said above, the uuid should be on the class itself instead of that of one of its children.